aten/src/ATen/mps/MPSStream.mm

```diff
@@ -232,18 +242,30 @@ @interface MPSGraphExecutionDescriptor ()
                     executionDescriptor:_executionDescriptor];
       }
 
+      updateCommandBufferResourceSize([feeds allValues]);
+      // if commitAndContinue is disabled, we need to always commit manually after encoding
+      SyncType _syncType = _enableCommitAndContinue == false ? SyncType::COMMIT : syncType;
+
       // check if graph execution profiling is enabled
       if (isGraphProfilingEnabled) {
-        // with profiler enabled, we commit after adding the completedHandler
-        profiler.endProfileKernel(mpsGraph, SyncType::COMMIT);
+        // with profiler enabled, we commit after adding the completedHandler in MPSProfiler
+        profiler.endProfileKernel(mpsGraph, _syncType);
       } else {
-        // if commitAndContinue is disabled, we need to always commit manually after encoding
-        synchronize(_enableCommitAndContinue == false ? SyncType::COMMIT : syncType);
+        synchronize(_syncType);
       }
     }
  });
 }
 
+void MPSStream::updateCommandBufferResourceSize(NSArray<MPSGraphTensorData*> *feeds) {
+  if (_enableCommitAndContinue) {
```

Comment:
What about resources from kernels ?

Reply:
There's no way to track them in a central place as the kernels are scattered everywhere. We can come back to them later.